src/OrbitGl/SchedulingStats.cpp

```diff
@@ -0,0 +1,89 @@
+                                 uint64_t end_ns) {
+  time_range_ms_ = static_cast<double>(end_ns - start_ns) * kNsToMs;
+
+  // Iterate on every scope in the selected range to compute stats.
+  for (const TextBox* scope : scheduling_scopes) {
+    const orbit_client_protos::TimerInfo& timer_info = scope->GetTimerInfo();
+    uint64_t clipped_start_ns = std::max(start_ns, timer_info.start());
+    uint64_t clipped_end_ns = std::min(end_ns, timer_info.end());
+    uint64_t timer_duration_ns = clipped_end_ns - clipped_start_ns;
+
+    time_on_core_ns_ += timer_duration_ns;
+    time_on_core_ns_by_core_[timer_info.processor()] += timer_duration_ns;
+
+    ProcessStats& process_stats = process_stats_by_pid_[timer_info.process_id()];
+    process_stats.time_on_core_ns_ += timer_duration_ns;
+
+    ThreadStats& thread_stats = process_stats.thread_stats_by_tid[timer_info.thread_id()];
+    thread_stats.time_on_core_ns_ += timer_duration_ns;
+  }
+
+  // Iterate on every process and thread to finalize stats.
+  for (auto& [pid, process_stats] : process_stats_by_pid_) {
+    process_stats.process_name = thread_name_provider(pid);
+    process_stats.pid = pid;
+    process_stats.time_on_core_ms = static_cast<double>(process_stats.time_on_core_ns_) * kNsToMs;
+    process_stats_sorted_by_time_on_core_.push_back(&process_stats);
+    for (auto& [tid, thread_stats] : process_stats.thread_stats_by_tid) {
+      thread_stats.thread_name = thread_name_provider(tid);
+      thread_stats.tid = tid;
+      thread_stats.time_on_core_ms = static_cast<double>(thread_stats.time_on_core_ns_) * kNsToMs;
+      process_stats.thread_stats_sorted_by_time_on_core.push_back(&thread_stats);
+    }
+  }
+
+  // Sort process stats by time on core.
+  std::sort(process_stats_sorted_by_time_on_core_.begin(),
+            process_stats_sorted_by_time_on_core_.end(), ProcessStatsTimeSorter());
+
+  // Sort thread stats by time on core.
+  for (ProcessStats* process_stats : process_stats_sorted_by_time_on_core_) {
+    std::sort(process_stats->thread_stats_sorted_by_time_on_core.begin(),
+              process_stats->thread_stats_sorted_by_time_on_core.end(), ThreadStatsTimeSorter());
+  }
+}
+
+std::string SchedulingStats::ToString() const {
+  std::string summary;
+
+  // Core occupancy.
+  if (time_on_core_ns_by_core_.size()) summary += "Core occupancy: \n";
+  for (auto& [core, time_on_core_ns_] : time_on_core_ns_by_core_) {
+    double time_on_core_ms = static_cast<double>(time_on_core_ns_) * kNsToMs;
+    summary +=
+        absl::StrFormat("cpu[%u] : %.2f%%\n", core, 100.0 * time_on_core_ms / time_range_ms_);
+  }
+
+  // Process and thread stats.
+  if (time_range_ms_ > 0) summary += absl::StrFormat("\nSelection time: %.6f ms\n", time_range_ms_);
+  for (ProcessStats* p_stats : process_stats_sorted_by_time_on_core_) {
+    summary += absl::StrFormat("  %s[%i] spent %.6f ms on core (%.2f%%)\n", p_stats->process_name,
```

Comment:
<img width="471" alt="selection time" src="https://user-images.githubusercontent.com/8610429/113991120-71dd4980-9852-11eb-8a1f-6168c10b8f04.png">

I would like to visually differentiate a little more between Process and Threads stats.

Maybe adding a "-" in every thread stats, to make the user visualize threads as a list from process?
```
for (ThreadStats* t_stats : p_stats->thread_stats_sorted_by_time_on_core) {
      summary += absl::StrFormat("  - %s[%i] spent %.6f ms on core (%.2f%%)\n"
```

Reply:
cool, done!